dev/ToastNotifications/ToastActivationCallback.h

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "NotificationActivationCallback.h"
+
+struct ToastActivationCallback : winrt::implements<ToastActivationCallback, INotificationActivationCallback>
+{
+    HRESULT __stdcall Activate(
+        LPCWSTR appUserModelId,
+        LPCWSTR invokedArgs,
+        [[maybe_unused]] NOTIFICATION_USER_INPUT_DATA const* data,
+        [[maybe_unused]] ULONG dataCount) noexcept;
+};
+
+struct ToastActivationCallbackFactory : winrt::implements<ToastActivationCallbackFactory, IClassFactory>
+{
+    STDMETHODIMP CreateInstance(_In_opt_ IUnknown* aggregateInterface, _In_ REFIID interfaceId, _Outptr_ VOID** object) noexcept final try
+    {
+        RETURN_HR_IF(CLASS_E_NOAGGREGATION, aggregateInterface != nullptr);
```

Comment:
RETURN_HR_IF_NULL

Reply:
Actually why do we need this check at all? This variable is a noop in any case

Reply:
We have the same check in PushNotificationBackgroundTask.h. Which is also an unused variable similar to this. To be honest I'm unsure what CLASS_E_NOAGGREGATION does. We can probably remove it.

Reply:
Callers that pass in a `aggregateInterface` are expecting an aggregate object (providing access to various inner and outer interfaces/methods). I think it's important you keep this in place to fulfill the contract, indicating to callers that aggregation is not supported.

See also: https://docs.microsoft.com/en-us/windows/win32/api/unknwn/nf-unknwn-iclassfactory-createinstance